purchase_analytic_global/models/purchase.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2014-2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import Command, api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    account_analytic_id = fields.Many2one(
+        "account.analytic.account",
+        string="Analytic Account",
+        compute="_compute_analytic_account",
+        inverse="_inverse_analytic_account",
+        help="This account will be propagated to all lines distributions, if you need "
+        "to use different distribution accounts, define the accounts at line level.",
+    )
```

Comment:
We want to add an analytic distribution here, and propagate it to all the lines. This is feasible by adding the analytic.mixin class (see https://github.com/odoo/odoo/blob/18.0/addons/analytic/models/analytic_mixin.py#L8) on the purchase.order model.